vispy/visuals/image_complex.py

```diff
@@ -0,0 +1,96 @@
+from .image import ImageVisual, _apply_clim_float, _apply_gamma_float
+import numpy as np
+from .shaders import Function, FunctionChain
+
+# In a complex Image, the texture will be rg32f, where:
+# data.r contains the real component
+# data.g contains the imaginary component
+COMPLEX_TRANSFORMS = {
+    "real": "float cplx2float(vec4 data) { return data.r; }",
+    "imaginary": "float cplx2float(vec4 data) { return data.g; }",
+    "magnitude": "float cplx2float(vec4 data) { return length(data); }",
+    "phase": "float cplx2float(vec4 data) { return atan(data.g, data.r); }",
+}
+# aliases
+COMPLEX_TRANSFORMS["angle"] = COMPLEX_TRANSFORMS["phase"]
+COMPLEX_TRANSFORMS["abs"] = COMPLEX_TRANSFORMS["magnitude"]
+COMPLEX_TRANSFORMS["imag"] = COMPLEX_TRANSFORMS["imaginary"]
+
+
+class ComplexImageVisual(ImageVisual):
```

Comment:
Docstring please. You could reference the ImageVisual directly at some point:

```
:class:`~vispy.visuals.ImageVisual`
```

Reply:
docstrings and tests coming still...